### PR TITLE
NETStandard 2.0 support

### PR DESCRIPTION
**Pull request by seanamos**
Issue: https://github.com/haf/Http.fs/issues/125

Public API of Http.fs remains the same.
Changed paket location to .paket (most things assume its here).
.fsproj files updated to the new format.
Compiles for net45 and netstandard2.0.
The unit tests and integration tests were updated to use Suave and Expecto.
Integration tests run on both a .netcoreapp2.0 (dotnet run) and a net461 EXE.
Rakefile updated to use dotnet cli and dotnet pack.

There are 2 integration tests I have commented out and marked with `FEEDBACK:` where behavior is either inconsistent between .NET frameworks or the original expectation did not pass:

 - `if KeepAlive is true, Connection set to 'Keep-Alive' on the first request, but not subsequent ones`
 - `cookies are kept during an automatic redirect`

TBD:
I haven't changed the travis / appveyor builds, so CI will fail. Probably going to need to do a few more changes to get things working properly on travis.

### CODE REVIEW

**Comment by haf** [comment]
Just rename from WIP when you feel that you're getting closer =)

File: HttpFs.IntegrationTests/Tests.fs [hunk 1]
```diff
@@ -117,7 +117,8 @@ let recorded =
       Expect.stringContains (req.Value |> getHeader "accept-language") "en-GB" "accept-language should be set"
       Expect.stringContains (req.Value |> getHeader "accept-language") "en-US" "accept-language should be set"
       Expect.equal (req.Value |> getHeader "authorization") "QWxhZGRpbjpvcGVuIHNlc2FtZQ==" "authorization should be equal"
-      Expect.stringContains (req.Value |> getHeader "connection") "conn1" "connection should be set"
+      // difference in behaviour between .NET/Core and mono. mono repos connection as "close"
```
Comment (haf) [inline HttpFs.IntegrationTests/Tests.fs]: ...and .net core?

**Comment by haf** [comment]
Wow @ your heavy lifting. I really appreciate the effort you're putting in. :)

File: HttpFs.IntegrationTests/Tests.fs (hunk 1 above)
Comment (seanamos) [reply HttpFs.IntegrationTests/Tests.fs]: Full .NET and Core have the same behavior. Mono seems to set the header to "close". I will play with it some more. It's either a difference in the underlying client behavior or the HTTP server (Suave).

**Comment by seanamos** [comment]
It's mostly getting the mono build to work on Travis. AppVeyor has been stable for some time now and the .NET Standard part of the build/tests has been working reliably as well, so I have fairly good confidence that NET 461/NET Core are good. I don't have a local mono linux setup so getting the CI to do the heavy lifting of checking it, often with some silly mistakes on my part.

For the most part the issues have been related to: https://github.com/dotnet/sdk/issues/335
All tests are passing with the exception of those mentioned above. Travis is just failing now on creating the nuget packages, again related to the SDK issue. There are some workarounds that I'm going to play with.

**Comment by seanamos** [comment]
There is an additional problem that will require a more extensive code change.
https://github.com/dotnet/corefx/issues/18237

In .NET Core HttpWebRequest is a wrapper around HttpClient. I looked further when I noticed the integration tests for .NET Core taking longer than those for .NET 461.

This boils down to the common HttpClient pitfall of not reusing the same HttpClient per host.
It can be addressed here, or addressed in a separate issue.

**Comment by seanamos** [comment]
^ I believe this also explains why this test fails on .NET Core:
`if KeepAlive is true, Connection set to 'Keep-Alive' on the first request, but not subsequent ones`

**Comment by haf** [comment]
I don't mind migrating to HttpClient, but in the long run we should have our own F# HTTP client that speaks TCP, IMO.

**Comment by haf** [comment]
It should be a separate PR btw.

**Comment by seanamos** [comment]
Finally green on CI. Was mostly a battle with the xplat SDK issues.

**Comment by haf** [comment]
Wow.

Thank you.

**Pull request merged**